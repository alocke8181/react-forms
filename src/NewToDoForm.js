import React, {useState} from "react";

const NewToDoForm = ({addTodo})=>{
    const [formData, setFormData] = useState({text:""});

    const handleChange = (e) =>{
        const {name, value} = e.target;
        setFormData((data) =>({
            ...data,
            [name]: value
        }));
    };

    const handleSubmit = (e)=>{
        e.preventDefault();
        addTodo(formData.text);
        setFormData({text:""});
    };

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="text">New Todo Item:</label>
            <input
                id="text"
                type="text"
                name="text"
                placeholder=""
                value={formData.text}
                onChange={handleChange}
            />
            <button id="todo-submit">Submit</button>
        </form>
    )

}

export default NewToDoForm