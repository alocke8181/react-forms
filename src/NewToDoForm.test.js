import { render } from '@testing-library/react';
import NewToDoForm from './NewToDoForm';

test('smoke',()=>{
    render(<NewToDoForm addTodo={console.log}/>)
})

test('snapshot',()=>{
    const {asFragment} = render(<NewToDoForm addTodo={console.log}/>)
    expect(asFragment()).toMatchSnapshot();
})