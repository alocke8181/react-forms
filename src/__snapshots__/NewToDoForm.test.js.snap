// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="text"
    >
      New Todo Item:
    </label>
    <input
      id="text"
      name="text"
      placeholder=""
      type="text"
      value=""
    />
    <button
      id="todo-submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
